projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-types/config-attribute-drop-down/config-attribute-drop-down.component.ts

```diff
@@ -5,29 +5,71 @@ import {
   Input,
   OnInit,
   Output,
+  OnDestroy,
 } from '@angular/core';
+import { formatNumber } from '@angular/common';
 import { FormControl } from '@angular/forms';
-import { Configurator } from '@spartacus/core';
+import { Configurator, LanguageService } from '@spartacus/core';
 import { ConfigFormUpdateEvent } from '../../config-form/config-form.event';
 import { ConfigUIKeyGeneratorService } from '../../service/config-ui-key-generator.service';
+import { Subscription } from 'rxjs';
+import { ConfigUtilsLocalizationService } from '../../service/config-localization-utils.service';
+
 @Component({
   selector: 'cx-config-attribute-drop-down',
   templateUrl: './config-attribute-drop-down.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfigAttributeDropDownComponent implements OnInit {
+export class ConfigAttributeDropDownComponent implements OnInit, OnDestroy {
   attributeDropDownForm = new FormControl('');
+  subscriptions = new Subscription();
+  locale: string;
 
-  constructor(public uiKeyGenerator: ConfigUIKeyGeneratorService) {}
+  constructor(
+    public uiKeyGenerator: ConfigUIKeyGeneratorService,
+    public languageService: LanguageService,
+    public utilsService: ConfigUtilsLocalizationService
+  ) {}
 
   @Input() attribute: Configurator.Attribute;
   @Input() group: string;
   @Input() ownerKey: string;
 
+  internalAttribute: Configurator.Attribute;
```

Comment:
'internalAttribute' is probably not a good name. The attribute is supposed to carry localized values, this should be reflected in the name